src/alchemlyb/parsing/lammps.py

```diff
@@ -0,0 +1,1303 @@
+        else:
+            dHdl = dHdl.mul({"coul": beta})
+
+    dHdl.name = "dH_dl"
+
+    return dHdl
+
+
+@_init_attrs
+def extract_H(
+    fep_files,
+    T,
+    column_lambda1=1,
+    column_pe=5,
+    column_lambda2=None,
+    units="real",
+    ensemble="nvt",
+    pressure=None,
+    column_volume=6,
+):
+    """Return reduced potentials Hamiltonian from LAMMPS dump file(s).
+
+    Each file is imported as a data frame where the columns kept are either::
+
+        [0, column_lambda, column_dlambda1, columns_derivative[0], columns_derivative[1]]
+
+    or if columns_lambda2 is not None::
+
+        [
+            0, column_lambda, column_dlambda1, column_lambda2, column_dlambda2,
+            columns_derivative1[0], columns_derivative1[1]
+        ]
+
+    Parameters
+    ----------
+    fep_files : str or list
+        If not a list, represents the path to fepout file(s) to extract data from. Filenames and paths are
+        aggregated using `glob <https://docs.python.org/3/library/glob.html>`_. For example, "/path/to/files/something_*_*.txt".
+    T : float
+        Temperature in Kelvin at which the simulation was sampled.
+    column_lambda1 : int, default=1
+        Index for column (column number minus one) representing the lambda at which the system is equilibrated.
+    column_pe : int, default=5
+        Index for column (column number minus one) representing the potential energy of the system.
+    column_lambda2 : int, default=None
+        Index for column (column number minus one) for a second value of lambda.
+        If this array is ``None`` then we do not expect two lambda values.
+    units : str, default="real"
+        Unit system used in LAMMPS calculation. Currently supported: "cgs", "electron", "lj". "metal", "micro", "nano",
+        "real", "si"
+    ensemble : str, default="nvt"
+        Ensemble from which the given data was generated. Either "nvt" or "npt" is supported where values from NVT are
+        unaltered, while those from NPT are corrected
+    pressure : float, default=None
+        The pressure of the system in the NPT ensemble in units of energy / volume, where the units of energy and volume
+        are as recorded in the LAMMPS dump file.
+    column_volume : int, default=4
+        The column for the volume in a LAMMPS dump file.
+
+    Results
+    -------
+    H : pandas.Dataframe
+        Dataframe of potential energy for each alchemical state (k) for each frame (n).
+        Note that the units for timestamps are not considered in the calculation.
+
+        Attributes
+
+        - temperature in K or dimensionless
+        - energy unit in kT
+
+    .. versionadded:: 2.4.1
+
+    """
+
+    # Collect Files
+    if isinstance(fep_files, list):
+        files = fep_files
+    else:
+        files = glob.glob(fep_files)
+    if not files:
+        raise ValueError("No files have been found that match: {}".format(fep_files))
+
+    if ensemble == "npt":
```

Comment:
Would be good if these check could be abstracted so you don't need to have the same thing in every function.